include/ensmallen_bits/ada_sqrt/ada_sqrt_impl.hpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file ada_sqrt_impl.hpp
+ * @author Marcus Edel
+ *
+ * Implementation of AdaSqrt optimizer.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ADA_SQRT_ADA_SQRT_IMPL_HPP
+#define ENSMALLEN_ADA_SQRT_ADA_SQRT_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "ada_sqrt.hpp"
```

Comment:
Is this `#include` necessary?

`ada_sqrt_impl.hpp` gets included from `ada_sqrt.hpp`, which is in turn included from `ensmallen.hpp`.

If I recall correctly, at the outset we decided not to support usage of only specific optimisers, as it's too messy (maintenance nightmare) and potentially buggy.  Only usage via `#include <ensmallen.hpp>` is supported.

Reply:
Not at all, it's an artifact from the mlpack code base, I'll remove it and also adapt existing optimizer in another PR.

Reply:
Yeah, I originally started doing that years ago to be nice to users who strangely included an `_impl.hpp` only. That makes more sense in the context of mlpack where you could choose multiple headers to include, but here in ensmallen it may not be needed, since users should only include the main ensmallen header.